app/src/main/java/com/obsidian/plugins/task_notifier/core/ObsidianTaskReminderCore.kt

```diff
@@ -28,6 +29,7 @@ class ObsidianTaskReminderCore {
                 return OnFileChangedResult.STOP_LISTENING;
```

Comment:
please add this block to handle scenarios when the folder should not be watched but we are still getting notifications from the running service:
```
if(!folders.contains(uri.path)) return OnFileChangedResult.STOP_LISTENING
```

Also inside the Service itself call stopService when the result of onFileChanged is STOP_LISTENING